apps/signup-form/src/components/IFrame.tsx

```diff
@@ -1,70 +1,58 @@
-/* eslint-disable @typescript-eslint/no-explicit-any */
-import {Component} from 'react';
+import React from 'react';
 import {createPortal} from 'react-dom';
 
-/**
- * This is still a class component because it causes issues with the behaviour (DOM recreation and layout glitches) if we switch to a functional component. Feel free to refactor.
- */
-export default class IFrame extends Component<any> {
-    node: any;
-    iframeHtml: any;
-    iframeHead: any;
-    iframeRoot: any;
-
-    constructor(props: {onResize?: (el: HTMLElement) => void, children: any}) {
-        super(props);
-        this.setNode = this.setNode.bind(this);
-        this.node = null;
-    }
-
-    componentDidMount() {
-        this.node.addEventListener('load', this.handleLoad);
-    }
-
-    handleLoad = () => {
-        this.setupFrameBaseStyle();
-    };
-
-    componentWillUnmount() {
-        this.node.removeEventListener('load', this.handleLoad);
-    }
-
-    setupFrameBaseStyle() {
-        if (this.node.contentDocument) {
-            this.iframeHtml = this.node.contentDocument.documentElement;
-            this.iframeHead = this.node.contentDocument.head;
-            this.iframeRoot = this.node.contentDocument.body;
-            this.forceUpdate();
-
-            if (this.props.onResize) {
-                // eslint-disable-next-line @typescript-eslint/no-unused-vars
-                (new ResizeObserver(_ => this.props.onResize(this.iframeRoot)))?.observe?.(this.iframeRoot);
+type IFrameProps = React.PropsWithChildren<{
+    style: React.CSSProperties
+    head: React.ReactNode;
+    title: string;
+    onResize?: (el: HTMLElement) => void;
+} & Omit<React.ComponentProps<'iframe'>, 'onResize'>>
+
+const IFrame: React.FC<IFrameProps> = ({children, head, title = '', onResize, style = {}, ...rest}) => {
+    const [iframeHead, setIframeHead] = React.useState<HTMLHeadElement>();
+    const [iframeRoot, setIframeRoot] = React.useState<HTMLElement>();
+
+    // TODO: Migrate to callback ref when React 19 cleanup refs are available: https://react.dev/blog/2024/04/25/react-19#cleanup-functions-for-refs
+    const node = React.useRef<HTMLIFrameElement>(null);
+    React.useEffect(() => {
+        function setupFrameBaseStyle() {
+            if (node.current?.contentDocument) {
+                const iframeRootLocal = node.current.contentDocument.body;
+                // This is safe because of batched updates, new to React 18
+                setIframeHead(node.current.contentDocument.head);
+                setIframeRoot(iframeRootLocal);
```

Comment:
No need to recall `this.forceUpdate` anymore, since the batched updates will trigger a re-render, updating the `iframeHead` and `iframeRoot` usages in the JSX